go/pkg/client/tree.go

```diff
@@ -366,15 +366,18 @@ func loadFiles(execRoot, localWorkingDir, remoteWorkingDir string, excl []*comma
 				return meta.Err
 			}
 
-			f, err := os.Open(absPath)
-			if err != nil {
-				return err
-			}
+			files := meta.DirChildren
+			if files == nil {
+				f, err := os.Open(absPath)
```

Comment:
This code seems repeated, can this be a utility function that tree.go and filemetadata.go uses?

Reply:
It's going to be:
```
f, err := os.Open(absPath)
if err != nil {
	return err
}

files, err = f.Readdirnames(-1)
f.Close()
if err != nil {
	return err
}
```
vs
```
import "github.com/bazelbuild/remote-apis-sdks/go/pkg/io"

names, err := io.Readdirnames(absPath)
if err != nil {
	return err
}
files = names
```
but with a new exported package just for the utility function. I'm hesitant about exporting helper functions.

Reply:
I do think the latter impl reduces cognitive load, but upto you though.

Reply:
I agree it's cleaner, but I'm not sure about the public helper function in a library. Tbh, many packages should have been internal.